Gems/LmbrCentral/Code/Source/Shape/EditorBoxShapeComponent.cpp

```diff
@@ -74,9 +74,10 @@ namespace LmbrCentral
             AZ::EntityComponentIdPair(GetEntityId(), GetId()));
 
         // ComponentMode
+        const bool allowAsymmetricalEditing = IsShapeComponentTranslationEnabled();
         m_componentModeDelegate.ConnectWithSingleComponentMode<
             EditorBoxShapeComponent, AzToolsFramework::BoxComponentMode>(
-                AZ::EntityComponentIdPair(GetEntityId(), GetId()), this);
+                AZ::EntityComponentIdPair(GetEntityId(), GetId()), this, allowAsymmetricalEditing);
```

Comment:
Is the feature flag for this going to be relatively short-lived? If so, it might make sense to query it down in the edit code instead of exposing it through a few layers of APIs. I realize this would have the side effect of making the ColliderBoxMode asymmetric editing wrapped with the feature flag too, but that seems like a pretty harmless consequence.

Reply:
Either way, I'm still in favor of the changes to the component mode delegate code that allows for passing down constructor parameters though. :)

Reply:
Yeah, the flag should be gone in a few weeks hopefully. As you mentioned, I did it that way because I didn't want to mix the logic for the collider (which can do asymmetrical editing whether the flag is on or not) and the shape components. And being able to pass the arguments through will probably be useful in general, so I went with that approach.

Reply:
I'm skeptical that exposing the asymmetric editing flag itself will be generally useful after a few weeks, but I don't feel that strongly about it, just raising the opinion for consideration. I've approved the PR either way. :)